tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -251,32 +250,75 @@ protected virtual bool ShouldEnableRemoteConfiguration(ImmutableTracerSettings s
+            // simply overriden with the remote configuration. Instead, remote rules are merged with local rules,
+            // with remote rules taking precedence.
+
             var sampler = new TraceSampler(new TracerRateLimiter(settings.MaxTracesSubmittedPerSecondInternal));
-            var samplingRules = settings.CustomSamplingRulesInternal;
-            var patternFormatIsValid = SamplingRulesFormat.IsValid(settings.CustomSamplingRulesFormat, out var samplingRulesFormat);
 
-            if (patternFormatIsValid && !string.IsNullOrWhiteSpace(samplingRules))
+            // remote sampling rules
+            var remoteSamplingRulesJson = settings.RemoteSamplingRules;
+
+            if (!string.IsNullOrWhiteSpace(remoteSamplingRulesJson))
             {
-                foreach (var rule in CustomSamplingRule.BuildFromConfigurationString(samplingRules, samplingRulesFormat, RegexBuilder.DefaultTimeout))
+                var remoteSamplingRules =
+                    RemoteCustomSamplingRule.BuildFromConfigurationString(remoteSamplingRulesJson, RegexBuilder.DefaultTimeout);
+
+                foreach (var remoteSamplingRule in remoteSamplingRules)
                 {
-                    sampler.RegisterRule(rule);
+                    sampler.RegisterRule(remoteSamplingRule);
                 }
             }
 
-            if (settings.GlobalSamplingRateInternal != null)
+            // local sampling rules
+            var patternFormatIsValid = SamplingRulesFormat.IsValid(settings.CustomSamplingRulesFormat, out var samplingRulesFormat);
+            var localSamplingRulesJson = settings.CustomSamplingRulesInternal;
+
+            if (patternFormatIsValid && !string.IsNullOrWhiteSpace(localSamplingRulesJson))
             {
-                var globalRate = (float)settings.GlobalSamplingRateInternal.Value;
+                var localSamplingRules =
+                    LocalCustomSamplingRule.BuildFromConfigurationString(localSamplingRulesJson, samplingRulesFormat, RegexBuilder.DefaultTimeout);
 
-                if (globalRate < 0f || globalRate > 1f)
+                foreach (var localSamplingRule in localSamplingRules)
                 {
-                    Log.Warning("{ConfigurationKey} configuration of {ConfigurationValue} is out of range", ConfigurationKeys.GlobalSamplingRate, settings.GlobalSamplingRateInternal);
+                    sampler.RegisterRule(localSamplingRule);
+                }
+            }
+
+            // global sampling rate (remote value overrides local value)
+            if (settings.GlobalSamplingRateInternal is { } globalSamplingRate)
+            {
+                if (globalSamplingRate is < 0f or > 1f)
+                {
+                    Log.Warning(
+                        "{ConfigurationKey} configuration of {ConfigurationValue} is out of range",
+                        ConfigurationKeys.GlobalSamplingRate,
+                        settings.GlobalSamplingRateInternal);
```

Comment:
Should we handle this in `TracerSettings` rather than here, where it should be recorded as an error in telemetry?

Reply:
Good catch. I added validation in `TracerSettings` (and removed the manual telemetry we had there for the default value, which was wrong because the default value is `null`, not `1`).

However, users can still set `TracerSettings.GlobalSamplingRate` in code, so I left the validation here. Is there a better way to handle this?

Reply:
I had to change a few tests for this because they expected the default value to be `1.0`. Not sure why the tests expected that, since the default value is `null` (= no global sampling rate). Defaulting to `1.0` would make it so we _never_ used the sampling rates from the agent.

Reply:
> I had to change a few tests for this because they expected the default value to be 1.0.

😅 

> However, users can still set TracerSettings.GlobalSamplingRate in code, so I left the validation here. Is there a better way to handle this?

If they're setting it in code, we could (and arguably should) be doing the validation there, and recording the changed value in telemetry in telemetry I think. Telemetry should include the values the values they have set, whether they are an are, _and_ the values we ended up using. Happy to move this to a separate PR/talk through how to implement it if you prefer!

Reply:
Scratch that. I removed all the changes related to `DD_TRACE_SAMPLE_RATE` since it causes a system test to fail (wrongly, I think). I've moved those commits to another branch.